llvm/lib/Target/X86/X86TargetTransformInfo.cpp

```diff
@@ -6894,3 +6894,82 @@ InstructionCost X86TTIImpl::getBranchMispredictPenalty() const {
   // TODO: Hook MispredictPenalty of SchedMachineModel into this.
   return 14;
 }
+
+static bool isVectorShiftByScalarCheap(Type *Ty, const X86Subtarget *ST) {
```

Comment:
Maybe move TargetLowering::isVectorShiftByScalarCheap into TTI as well? X86 is its only user

Reply:
Sure, I've moved it, though I think the function needs a minor rework if we wanted to use the cost model as its implementation.